x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/enrich/EnrichPolicy.java

```diff
@@ -141,6 +164,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.array(INDICES.getPreferredName(), indices.toArray(new String[0]));
         builder.field(ENRICH_KEY.getPreferredName(), enrichKey);
         builder.array(ENRICH_VALUES.getPreferredName(), enrichValues.toArray(new String[0]));
+        if (versionCreated != null) {
```

Comment:
Perhaps we should only serialize `versionCreated` if params.include_version is set to `true`? It should default to true, and in get and list policy api we would then set it to false. This way when the user adds a policy and then returns it via get policy api then he sees the exact content that was submitted.

Reply:
I've made this change, but it looks like `Strings.toString(this)` does not use `include_version` and can lead to some strange assert messages when checking equals on two policy objects where one has a version set and the other does not: 

```
    java.lang.AssertionError: 
    Expected: <{"type":"exact_match","indices":["index"],"enrich_key":"key","enrich_values":["val1"]}>
         but: was <{"type":"exact_match","indices":["index"],"enrich_key":"key","enrich_values":["val1"]}>
```

I don't think this is necessarily bad, but it can be confusing.

Reply:
Agreed this can be confusing, but I think it is ok. This things are easy to spot in debug mode.